store/copr/coprocessor.go

```diff
@@ -164,20 +195,17 @@ func (c *CopClient) Send(ctx context.Context, req *kv.Request, variables interfa
 				}
 			})
 		}
-		it.sendRate = util.NewRateLimit(2 * it.concurrency)
+		if it.smallTaskConcurrency > 20 {
+			it.smallTaskConcurrency = 20
+		}
```

Comment:
1. What happens if user set tidb_distsql_scan_concurrency bigger than 20?  Will it cause the SQL latency to be worse than before the feature was implemented?
2. If 20 is ok, why the default value 15 can not work well?

Reply:
> Will it cause the SQL latency to be worse than before the feature was implemented?

The concurrency is adjusted in keep order mode, before the feature, concurrency will be reduced to 2 in this case and it's not affected by `tidb_distsql_scan_concurrency`, see [description](https://github.com/pingcap/tidb/blob/84fbfcada3d9564fe922771f010db318d27fc77a/store/copr/coprocessor.go#L149-L158).

> If 20 is ok, why the default value 15 can not work well?

This number is not carefully chosen, just set an upper limit.